src/riak_ql_lexer.xrl

```diff
@@ -153,37 +153,78 @@ get_tokens(X) ->
 post_process(X) ->
     post_p(X, []).
 
-% filter out the whitespaces at the end
+%% filter out the whitespaces at the end
 post_p([], Acc) ->
     [{Type, X} || {Type, X} <- lists:reverse(Acc), Type =/= whitespace];
-% when you've merged two items hoy them back on the list
-% so they can continue to sook up chars
-post_p([{Word1, W1}, {Word2, W2} | T], Acc) when Word1 =:= chars   orelse
-                                                 Word1 =:= select  orelse
-                                                 Word1 =:= from    orelse
-                                                 Word1 =:= limit   orelse
-                                                 Word1 =:= and_    orelse
-                                                 Word1 =:= or_     orelse
-                                                 Word1 =:= delete  orelse
-                                                 Word1 =:= drop    orelse
-                                                 Word1 =:= groupby orelse
-                                                 Word1 =:= merge   orelse
-                                                 Word1 =:= inner   orelse
-                                                 Word1 =:= join    orelse
-                                                 Word1 =:= as,
-                                                 Word2 =:= chars   orelse
-                                                 Word2 =:= select  orelse
-                                                 Word2 =:= from    orelse
-                                                 Word2 =:= limit   orelse
-                                                 Word2 =:= and_    orelse
-                                                 Word2 =:= or_     orelse
-                                                 Word2 =:= delete  orelse
-                                                 Word2 =:= drop    orelse
-                                                 Word2 =:= groupby orelse
-                                                 Word2 =:= merge   orelse
-                                                 Word2 =:= inner   orelse
-                                                 Word2 =:= join    orelse
-                                                 Word2 =:= as      ->
+%% when you've merged two items hoy them back on the list
+%% so they can continue to sook up chars
+%% NOTE this when clause is generated by the key word generator
```

Comment:
I have made a change in the ddl_dist branch so that integer literals inside and after words are handled correctly e.g. `hi5`, h5i` which was causing an error. This means that if Word2 is an int atom then convert W2 to a binary and append it to W1.

Can the atom int be added to the Word2 guard only in the generator?